### PR TITLE
Add validation for threat intel source config

### LINKED ISSUE #1366: [BUG] PUT on source without "name" in request makes the source unable to be fetched/deleted/listed/updated
**What is the bug?**
Threat-intel Sources when modified using the API with PUT, without having the "name" in request body, makes the server throw 500 for every request that includes the source in the response, and the source can also not be DELETED

**How can one reproduce the bug?**
1. Utilize the API
2. Run POST _plugins/_security_analytics/threat_intel/sources (I utilized file_upload API) 
3. Run PUT  _plugins/_security_analytics/threat_intel/sources/<source_id> (Without "name" in request body")
4. Response Body: {"ok":false,"error":"[illegal_state_exception] Can't get text on a VALUE_NULL at 1:26"}

**What is the expected behavior?**
There should be a validation for this, and in case not, the source should still have a way to be deleted. Via the dashboards or API, and it should not break the other endpoints which return this ( Fetch All )

**What is your host/environment?**
 - Plugins

**Do you have any screenshots?**
Error 1
error 2

**Do you have any additional context?**
Crucially, the DELETE isn't working, so I cannot find a way to get rid of the faulty source via the API.

### PR DESCRIPTION
**Pull request by jowg-amazon**
### Description
Add source config validation so source config cannot be created without valid `name`, `type`, `format`, `source`, and `ioc type`. Additionally allows source configs with null values to still be read and deleted.

### Related Issues
Resolves #1366

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security-analytics/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by eirsep — APPROVED** [review]

**Review by eirsep — CHANGES_REQUESTED** [review]
can we add test to verify https://github.com/opensearch-project/security-analytics/issues/1366 is solved

File: src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java [hunk 1]
```diff
@@ -24,7 +24,19 @@ public class SourceConfigDtoValidator {
     public List<String> validateSourceConfigDto(SATIFSourceConfigDto sourceConfigDto) {
```
Comment (AWSHurneyt) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java:24]: As a follow-up item, let's add validation for the refresh frequency.

File: src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java [hunk 2]
```diff
@@ -24,7 +24,19 @@ public class SourceConfigDtoValidator {
...
         List<String> errorMsgs = new ArrayList<>();
 
-        if (sourceConfigDto.getIocTypes().isEmpty()) {
+        if (sourceConfigDto.getName() == null || sourceConfigDto.getName().isEmpty()) {
```
Comment (AWSHurneyt) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java:27]: @jowg-amazon could we add length restrictions? The frontend uses this validation for names.
https://github.com/opensearch-project/security-analytics-dashboards-plugin/blob/main/public/utils/validation.ts#L7C14-L7C33

File: src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java [hunk 3]
```diff
@@ -24,7 +24,19 @@ public class SourceConfigDtoValidator {
...
+            errorMsgs.add("Name must not be empty");
+        }
+
+        if (sourceConfigDto.getFormat() == null || sourceConfigDto.getFormat().isEmpty()) {
```
Comment (AWSHurneyt) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java:31]: We may need to add length validation for this as well. Unless this gets validated somewhere downstream, the config could be created via backend API with something like 300k characters; which could disrupt functionality.

File: src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java (hunk 2 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java:32]: added validation for names based on how frontend currently validates the name

File: src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java (hunk 3 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/common/SourceConfigDtoValidator.java:38]: Added length validation

**Comment by jowg-amazon** [comment]
> can we add test to verify #1366 is solved

Added parsing tests to make sure null fields can be parsed

**Review by AWSHurneyt — APPROVED** [review]
Changes LGTM. Approving assuming other's comments are addressed.

**Review by eirsep — APPROVED** [review]

**Pull request merged**